workstreams/Opaque_Pointers_Workstream.md

```diff
@@ -0,0 +1,98 @@
+# Workstream: Opaque Pointer Support
+
+## Motivation & Benefits
+
+Support for typed pointers in LLVM is being removed. More information about the
+motivation for this change can be found in LLVM forums. For example, the LLVM
+documentation on [opaque pointers](https://llvm.org/docs/OpaquePointers.html) as
+well as [this
+presentation](https://llvm.org/devmtg/2022-04-03/slides/keynote.Opaque.Pointers.Are.Coming.pdf)
+give a brief overview.
+
+The QIR specification currently makes use of typed pointers to represent qubit
+values and result values. While in LLVM 15, support for these types can still be
+enabled, an update to LLVM 16 requires us to change this representation.
+
+The goal of this workstream is hence to update the QIR specification to support
+newer versions of LLVM, and to document what LLVM versions can be used to
+support each version of the QIR specification.
+
+## Requirements
+
+Generally, LLVM maintains very good [backwards
+compatibility](https://llvm.org/docs/DeveloperPolicy.html#ir-backwards-compatibility);
+according to the LLVM 18 docs, it is possible to load even bitcode generate by
+an LLVM version as early as 3.0. However, it is not possible in general to
+generate bitcode that can be loaded with an older version of LLVM than the one
+used to emit it. This means that while *QIR consumers*, i.e. backend providers,
+can update to the latest LLVM version without sacrificing support for prior QIR
+versions, whereas *QIR producers*, i.e. language frontends, cannot use LLVM
+versions that are newer than the one required by the versions of the QIR
+specification they support. This can pose challenges for maintaining a
+state-of-the art frontend.
+
+As part of this workstream, we hence specifically propose to update the required
+LLVM version for the next version of the QIR specification to LLVM 16.
```

Comment:
Looks like LLVM 17 will be the 'latest' release by the time this workstream starts (release is planned for Sept 19th). Would it make sense to always target 'latest' in any 'move LLVM forward' type workstreams to maximize time before we have to migrate again?

Reply:
updated to llvm 17